cmd/lncli/main.go

```diff
@@ -73,28 +71,32 @@ func getClientConn(ctx *cli.Context) *grpc.ClientConn {
 			fatal(err)
 		}
 
-		// We add a time-based constraint to prevent replay of the
-		// macaroon. It's good for 60 seconds by default to make up for
-		// any discrepancy between client and server clocks, but leaking
-		// the macaroon before it becomes invalid makes it possible for
-		// an attacker to reuse the macaroon. In addition, the validity
-		// time of the macaroon is extended by the time the server clock
-		// is behind the client clock, or shortened by the time the
-		// server clock is ahead of the client clock (or invalid
-		// altogether if, in the latter case, this time is more than 60
-		// seconds).
-		// TODO(aakselrod): add better anti-replay protection.
-		macaroonTimeout := time.Duration(ctx.GlobalInt64("macaroontimeout"))
-		requestTimeout := time.Now().Add(time.Second * macaroonTimeout)
-		timeCaveat := checkers.TimeBeforeCaveat(requestTimeout)
-		mac.AddFirstPartyCaveat(timeCaveat.Condition)
+		macConstraints := []macaroons.Constraint{
+			// We add a time-based constraint to prevent replay of the
+			// macaroon. It's good for 60 seconds by default to make up for
+			// any discrepancy between client and server clocks, but leaking
+			// the macaroon before it becomes invalid makes it possible for
+			// an attacker to reuse the macaroon. In addition, the validity
+			// time of the macaroon is extended by the time the server clock
+			// is behind the client clock, or shortened by the time the
+			// server clock is ahead of the client clock (or invalid
+			// altogether if, in the latter case, this time is more than 60
+			// seconds).
+			// TODO(aakselrod): add better anti-replay protection.
+			macaroons.TimeoutConstraint(ctx.GlobalInt64("macaroontimeout")),
+
+			// Lock macaroon down to a specific IP address.
+			macaroons.IPLockConstraint(ctx.GlobalString("macaroonip")),
```

Comment:
I'd prefer for this to be either a) conditionally added if the macaroon IP is specified, or b) defaulted to `127.0.0.1`. I think I prefer the first. @Roasbeef, what's your preference?

Reply:
@Roasbeef agreed with me on this one.

Reply:
`macaroons.IPLockConstraint` does nothing if its argument is an empty string, which is the default value for the `-macaroonip` option, so it should work exactly as you specified.

Reply:
Ah, yes, you're correct. Thanks!